containers/daap-delete-data-product/tests/unit/mock_test.py

```diff
@@ -37,15 +37,25 @@
 }
 
 
+def count_files(client, bucket, prefix):
+    response = client.list_objects_v2(
+        Bucket=bucket,
+        Prefix=prefix,
+    )
+    return response.get("KeyCount")
```

Comment:
```suggestion
    # TODO: pagination if >1000 objects expected in bucket
    return response.get("KeyCount")
```

NOTE: as I understand, this has a [limit up to 1000](https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/s3/client/list_objects_v2.html) (and so [requires pagination](https://stackoverflow.com/a/70716816/10478115) if more than 1000 entries)

Reply:
Agreed, now paginated.